controllers/helmrepository_controller.go

```diff
@@ -209,11 +209,16 @@ func (r *HelmRepositoryReconciler) reconcile(ctx context.Context, repository sou
 		return sourcev1.HelmRepositoryNotReady(repository, sourcev1.IndexationFailedReason, err.Error()), err
 	}
 
-	// return early on unchanged generation
+	indexBytes, err := yaml.Marshal(chartRepo.Index)
```

Comment:
```suggestion
	indexBytes, err := yaml.Marshal(&chartRepo.Index)
```

This will be the last requested change, promised :raised_hand_with_fingers_splayed:

Reply:
Good catch, here we go!